src/lcm/error_handling.py

```diff
@@ -0,0 +1,77 @@
+from collections.abc import Callable
+
+import jax.numpy as jnp
+import pandas as pd
+from jax import Array
+
+from lcm._config import get_log_dir
+from lcm.exceptions import InvalidValueFunctionError
+from lcm.interfaces import StateActionSpace
+
+
+def validate_value_function_array_integrity(
+    V_arr: Array,
+    state_action_space: StateActionSpace,
+) -> None:
+    """Validate the value function array for NaN and infinity values.
+
+    This function checks the value function array for any NaN or infinity values. If any
+    such values are found, we log the corresponding state-combinations to
+
+    Args:
+        V_arr: The value function array to validate.
+        state_action_space: The state-action space to check against.
+
+    Raises:
+        InvalidValueFunctionError: If the value function array contains NaN or infinity
+            values. The invalid states are logged to CSV files in the '.pylcm' directory
+            for further analysis.
+
+    """
+    # Early return if the value function array contains only valid values
+    if not jnp.any(jnp.isnan(V_arr) | jnp.isinf(V_arr)):
+        return
+
+    nan_states_df = _get_dataframe_of_invalid_states(
+        V_arr=V_arr,
+        check_fn=jnp.isnan,
+        ordered_state_names=list(state_action_space.states),
+        states=state_action_space.states,
+    )
+
+    inf_states_df = _get_dataframe_of_invalid_states(
+        V_arr=V_arr,
+        check_fn=jnp.isinf,
+        ordered_state_names=list(state_action_space.states),
+        states=state_action_space.states,
+    )
```

Comment:
Lets combine this into a single check, single `invalid_states_df` and single CSV file.